mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -45,37 +48,63 @@ pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // Get all enabled tokens with platform coin including the coin.
+    let coins_to_disable = coins_ctx
+        .get_tokens_to_disable(&ticker)
+        .await
+        .into_iter()
+        .chain(std::iter::once(ticker.clone()))
+        .collect::<Vec<_>>();
+
+    // Get all matching orders and active swaps.
+    let mut active_swaps = vec![];
+    let mut still_matching_orders = vec![];
+    for ticker in &coins_to_disable {
+        let swaps = try_s!(active_swaps_using_coin(&ctx, ticker));
+        let still_matching = get_matching_orders(&ctx, ticker).await;
+        active_swaps.extend(swaps);
+        still_matching_orders.extend(still_matching);
     }
-    let (cancelled, still_matching) = try_s!(cancel_orders_by(&ctx, CancelBy::Coin { ticker: ticker.clone() }).await);
-    if !still_matching.is_empty() {
+    drop_mutability!(active_swaps);
+    drop_mutability!(still_matching_orders);
+    // If there're matching orders or active swap we return this error.
+    if !active_swaps.is_empty() || !still_matching_orders.is_empty() {
         let err = json!({
-            "error": format!("There're currently matching orders using {}", ticker),
+            "error": format!("There're currently matching orders or active swaps for some tokens"),
             "orders": {
-                "matching": still_matching,
-                "cancelled": cancelled,
+                "active_swaps": still_matching_orders,
+                "still_matching": still_matching_orders
             }
         });
         return Response::builder()
-            .status(500)
+            .status(INTERNAL_SERVER_ERROR_CODE)
             .body(json::to_vec(&err).unwrap())
             .map_err(|e| ERRL!("{}", e));
     }
 
-    try_s!(disable_coin_impl(&ctx, &ticker).await);
+    // Proceed with diabling the coin/tokens.
+    let mut disabled_tokens = vec![];
+    for ticker in &coins_to_disable {
+        log!("disabling {ticker} coin");
+        if let Err(err) = coins_ctx.remove_coin(&ctx, ticker).await {
+            let err = json!({
+                "error": err,
+                "disabled_tokens": disabled_tokens,
+            });
+            return Response::builder()
+                .status(INTERNAL_SERVER_ERROR_CODE)
+                .body(json::to_vec(&err).unwrap())
+                .map_err(|e| ERRL!("{}", e));
+        };
+        disabled_tokens.push(ticker);
+    }
+
     let res = json!({
         "result": {
             "coin": ticker,
-            "cancelled_orders": cancelled,
+            "disabled_tokens": disabled_tokens
```

Comment:
Why was the cancelling of orders removed? We should still cancel non matching orders for the tokens if there is no active swaps or still matching orders.